### PR TITLE
InlineForm component

### LINKED ISSUE #419: InlineForm component

### PR DESCRIPTION
**Pull request by Katrin-kudryash**
## PR includes
- [x] Feature

## Related issues
Resolve #419

### CODE REVIEW

**Review by awinogradov — CHANGES_REQUESTED** [review]
There no screenshots, videos, Storybook examples

**Comment by Katrin-kudryash** [comment]
Untitled 2

File: src/components/InlineForm.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                            <>
+                                <a
+                                    ref={hintRef}
+                                    onMouseOver={() => toggleHintVisible(true)}
+                                    onMouseLeave={() => toggleHintVisible(false)}
+                                >
+                                    <StyledQuestionIcon size="s" />
+                                </a>
+                                <Popup reference={hintRef} visible={hintVisible} placement="top-end">
+                                    {t}
```
Comment (9teen90nine) [inline src/components/InlineForm.tsx:113]: What is that?

Comment (Katrin-kudryash) [reply src/components/InlineForm.tsx:113]: text

File: src/components/InlineForm.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    tip?: React.ReactNode;
+    onReset: () => void;
+    isSubmitted?: boolean;
+    className?: string;
+}
+
+const StyledWrapper = styled.div`
+    display: contents;
+
+    & form {
```
Comment (awinogradov) [inline src/components/InlineForm.tsx:29]: Use styled-component instead `& ${StyledForm} { }`

File: src/components/InlineForm.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    onClick: () => void;
+}
+
+interface InlineFormProps {
+    renderTrigger?: (props: RenderTriggerProps) => React.ReactNode;
+    onSubmit: () => void;
+    children: React.ReactNode;
+    tip?: React.ReactNode;
+    onReset: () => void;
+    isSubmitted?: boolean;
```
Comment (awinogradov) [inline src/components/InlineForm.tsx:22]: What the reason to have this prop? Why onSubmit callback is not enough?

File: src/components/InlineForm.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        }
+
+        return null;
+    }, [renderTrigger]);
+
+    const [onESC] = useKeyboard(
+        [KeyCode.Escape],
+        () => {
+            if (visible) {
+                toggle();
```
Comment (awinogradov) [inline src/components/InlineForm.tsx:69]: `toggleVisible` much better

File: src/components/InlineForm.tsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    }, [visible, onReset]);
+
+    return (
+        <StyledWrapper ref={wrapperRef} className={className} {...onESC}>
+            {!visible && trigger}
+            {visible && (
+                <Form onSubmit={onSubmit}>
+                    <StyledFormWrapper>
+                        {children}
+                        {nullable(tip, (t: React.ReactNode) => (
```
Comment (awinogradov) [inline src/components/InlineForm.tsx:103]: Whole condition and tip prop must be removed

**Review by awinogradov — CHANGES_REQUESTED** [review]

**Comment by Katrin-kudryash** [comment]
Untitled 6

File: src/components/InlineForm.tsx (hunk 3 above)
Comment (Katrin-kudryash) [reply src/components/InlineForm.tsx]: ok, onSubmit callback is enough

File: src/components/InlineForm.tsx (hunk 1 above)
Comment (Katrin-kudryash) [reply src/components/InlineForm.tsx]: tip, I removed it from the form

File: src/components/InlineForm.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+interface InlineFormProps {
+    renderTrigger?: (props: RenderTriggerProps) => React.ReactNode;
+    onSubmit: () => Promise<void>;
+    children: React.ReactNode;
+    onReset: () => void;
+    className?: string;
+}
+
+const StyledForm = styled(Form)``;
```
Comment (awinogradov) [inline src/components/InlineForm.tsx:21]: Does it makes sense?

File: src/components/InlineForm.tsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        }
+    }, [visible, onReset]);
+
+    return (
+        <StyledWrapper ref={wrapperRef} className={className} {...onESC}>
+            {!visible && trigger}
+            {visible && (
+                <StyledForm
+                    onSubmit={() =>
+                        onSubmit().then(() => {
```
Comment (awinogradov) [inline src/components/InlineForm.tsx:78]: useCallback with await much better

File: src/stories/InlineForm.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+
+const StyledWrapper = styled.div`
+    display: contents;
+
+    & form {
+        background-color: transparent;
+        display: contents;
```
Comment (awinogradov) [inline src/stories/InlineForm.stories.tsx:28]: Why this property so popular in this PR. Does it really must be everywhere?

File: src/stories/InlineForm.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+    box-sizing: border-box;
+`;
+
+const StyledQuestionIcon = styled.div`
+    margin: 5px 0 0 4px;
+`;
+
+export const InlineFormComponent: Story = () => {
+    const handleFormInputChange = useCallback(() => {
+        console.log('Change Click');
```
Comment (awinogradov) [inline src/stories/InlineForm.stories.tsx:59]: Use storybook action instead.

File: src/stories/InlineForm.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+                            </Popup>
+                        </>
+                    </StyledTableCell>
+                </StyledFormWrapper>
+            </Form>
+        </StyledWrapper>
+    );
+};
+
+InlineFormComponent.args = {};
```
Comment (awinogradov) [inline src/stories/InlineForm.stories.tsx:100]: ?

**Review by awinogradov — CHANGES_REQUESTED** [review]

**Comment by Katrin-kudryash** [comment]
p

File: src/components/InlineForm.tsx [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+    useEffect(() => {
+        if (!visible) {
+            onReset();
+        }
+    }, [visible, onReset]);
+
+    return (
+        <StyledWrapper ref={wrapperRef} className={className} {...onESC}>
+            {!visible && trigger}
```
Comment (awinogradov) [inline src/components/InlineForm.tsx:73]: nullable

File: src/components/InlineForm.tsx [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+            {visible && <StyledForm onSubmit={handleSubmit}>{children}</StyledForm>}
```
Comment (awinogradov) [inline src/components/InlineForm.tsx:74]: nullable

File: src/stories/InlineForm.stories.tsx [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+`;
+
+const StyledTableCell = styled(TableCell)`
+    flex-wrap: nowrap;
+    display: flex;
+    align-items: center;
+`;
+
+const StyledFormInput = styled(FormInput)`
+    font-size: 14px;
```
Comment (awinogradov) [inline src/stories/InlineForm.stories.tsx:36]: does we really need this custom styles to make form example?

File: src/stories/InlineForm.stories.tsx [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+                <StyledTableCell col={5} align="center">
+                    <StyledFormInput autoFocus brick="right" placeholder={'Enter criteria'} />
+                    <StyledFormInput autoComplete="off" placeholder={'No more than 100'} brick="center" />
+                    <Button
+                        text={'Add'}
+                        brick="left"
+                        type="submit"
+                        view="primary"
+                        outline
+                        onClick={() => console.log('clecked')}
```
Comment (awinogradov) [inline src/stories/InlineForm.stories.tsx:62]: use storybook actions instead

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: src/stories/InlineForm.stories.tsx (hunk 13 above)
Comment (Katrin-kudryash) [reply src/stories/InlineForm.stories.tsx]: Not needed. We can use default

File: src/components/InlineForm.tsx [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+    onClick: () => void;
+}
+
+interface InlineFormProps {
+    renderTrigger: (props: RenderTriggerProps) => React.ReactNode;
+    onSubmit: () => Promise<void>;
+    children: React.ReactNode;
+    onReset: () => void;
+    className?: string;
+    text?: string;
```
Comment (awinogradov) [inline src/components/InlineForm.tsx:20]: Looks like this prop is not using anywhere

File: src/stories/InlineForm.stories.tsx [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+        onChange: { action: 'onChange' },
+    },
+} as Meta<typeof InlineForm>;
+
+const StyledWrapper = styled.div`
+    display: contents;
+`;
+
+const StyledForm = styled(Form)`
+    background-color: transparent;
```
Comment (awinogradov) [inline src/stories/InlineForm.stories.tsx:31]: this is by default

File: src/stories/InlineForm.stories.tsx [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+        onSubmit: { action: 'form submitted' },
+        onClick: { action: 'clecked' },
+        onMouseOver: { action: 'onMouseOver' },
+        onMouseLeave: { action: 'onMouseLeave' },
+        onChange: { action: 'onChange' },
+    },
+} as Meta<typeof InlineForm>;
+
+const StyledWrapper = styled.div`
+    display: contents;
```
Comment (awinogradov) [inline src/stories/InlineForm.stories.tsx:27]: U already have this wrapper in InlineForm component. U are wrapping form twice in the same component.

File: src/stories/InlineForm.stories.tsx [hunk 18]
```diff
@@ -0,0 +1,71 @@
...
+
+const StyledQuestionIcon = styled.div`
+    margin: 5px 0 0 4px;
+`;
+
+export const InlineFormStories: StoryFn = () => {
+    return (
+        <StyledWrapper>
+            <StyledForm onSubmit={action('form submitted')}>
+                <StyledTableCell col={5} align="center">
```
Comment (awinogradov) [inline src/stories/InlineForm.stories.tsx:48]: I'm not sure we need this part of table here

**Review by awinogradov — CHANGES_REQUESTED** [review]

**Review by 9teen90nine — APPROVED** [review]

**Review by awinogradov — APPROVED** [review]

**Pull request merged**